vespa-feed-client-cli/src/main/java/ai/vespa/feed/client/CliClient.java

```diff
@@ -60,22 +65,55 @@ private int run(String[] rawArgs) {
             try (InputStream in = createFeedInputStream(cliArgs);
                  FeedClient feedClient = createFeedClient(cliArgs);
                  JsonFeeder feeder = createJsonFeeder(feedClient, cliArgs)) {
+                CountDownLatch latch = new CountDownLatch(1);
+                AtomicReference<FeedException> fatal = new AtomicReference<>();
                 long startNanos = System.nanoTime();
-                feeder.feedMany(in).join();
+                feeder.feedMany(in, new ResultCallback() {
+                    @Override public void onNextResult(Result result, FeedException error) { handleResult(result, error, cliArgs); }
+                    @Override public void onError(FeedException error) { fatal.set(error); }
```

Comment:
Should latch count down be invoked `onError`? `onComplete` will not be invoked in this case.

Reply:
Oh, thne it definiteely should :)